addons/mail/static/tests/core/search_highlight_test.js

```diff
@@ -0,0 +1,84 @@
+/* @odoo-module */
+
+import { HIGHLIGHT_CLASS, searchHighlight } from "@mail/core/common/message_search_hook";
+
+QUnit.module("Search highlight test", {});
+
+QUnit.test("Search highlight", async (assert) => {
+    const testCases = [
+        {
+            input: "test odoo",
+            output: `test <span class="${HIGHLIGHT_CLASS}">odoo</span>`,
+            searchTerm: "odoo",
+        },
+        {
+            input: '<a href="https://www.odoo.com">https://www.odoo.com</a>',
+            output: `<a href="https://www.odoo.com">https://www.<span class="${HIGHLIGHT_CLASS}">odoo</span>.com</a>`,
+            searchTerm: "odoo",
+        },
+        {
+            input: '<a href="https://www.odoo.com">Odoo</a>',
+            output: `<a href="https://www.odoo.com"><span class="${HIGHLIGHT_CLASS}">Odoo</span></a>`,
+            searchTerm: "odoo",
+        },
+        {
+            input: '<a href="https://www.odoo.com">Odoo</a> Odoo is a free software',
+            output: `<a href="https://www.odoo.com"><span class="${HIGHLIGHT_CLASS}">Odoo</span></a> <span class="${HIGHLIGHT_CLASS}">Odoo</span> is a free software`,
+            searchTerm: "odoo",
+        },
+        {
+            input: "odoo is a free software",
+            output: `<span class="${HIGHLIGHT_CLASS}">odoo</span> is a free software`,
+            searchTerm: "odoo",
+        },
+        {
+            input: "software ODOO is a free",
+            output: `software <span class="${HIGHLIGHT_CLASS}">ODOO</span> is a free`,
+            searchTerm: "odoo",
+        },
+        {
+            input: `<ul>
+                <li>Odoo</li>
+                <li><a href="https://odoo.com">Odoo ERP</a> Best ERP</li>
+            </ul>`,
+            output: `<ul>
+                <li><span class="${HIGHLIGHT_CLASS}">Odoo</span></li>
+                <li><a href="https://odoo.com"><span class="${HIGHLIGHT_CLASS}">Odoo</span> ERP</a> Best ERP</li>
+            </ul>`,
+            searchTerm: "odoo",
+        },
+        {
+            input: "test <strong>Odoo</strong> test",
+            output: `<span class="${HIGHLIGHT_CLASS}">test</span> <strong><span class="${HIGHLIGHT_CLASS}">Odoo</span></strong> <span class="${HIGHLIGHT_CLASS}">test</span>`,
+            searchTerm: "odoo test",
+        },
+        {
+            input: "test <br> test",
+            output: `<span class="${HIGHLIGHT_CLASS}">test</span> <br> <span class="${HIGHLIGHT_CLASS}">test</span>`,
+            searchTerm: "odoo test",
+        },
+        {
+            input: "<strong>test</strong> test",
+            output: `<strong><span class="${HIGHLIGHT_CLASS}">test</span></strong> <span class="${HIGHLIGHT_CLASS}">test</span>`,
+            searchTerm: "test",
+        },
+        {
+            input: "<strong>a</strong> test",
+            output: `<strong><span class="${HIGHLIGHT_CLASS}">a</span></strong> <span class="${HIGHLIGHT_CLASS}">test</span>`,
+            searchTerm: "a test",
+        },
```

Comment:
missing tests for these situations where tags (strong, ...) are initially escaped in input and stay escaped in output

Reply:
Side note: I introduced more test to have a better coverage of the highlight feature